src/providers/contact.provider.ts

```diff
@@ -1,53 +1,34 @@
-import { HttpClient, HttpHeaders, HttpResponse } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-import { ApiGateway } from 'src/api-gateway';
-import { environment } from 'src/environments/environment';
+import { ApiClient } from 'src/apiClient';
+import { Contact } from 'src/models/contactModel';
+import { ApiContactResponse, ApiContactsResponse, ApiDeletedContact, ApiResponse } from 'src/utils/api-response';
+import { ApiTokenResponse } from 'src/utils/api-response';
 
 @Injectable({
   providedIn: 'root',
 })
 export class ContactProvider {
-  constructor(private apiGateway: ApiGateway, private http: HttpClient) { }
+  constructor(private apiClient: ApiClient) { }
 
   ngOnInit(): void { }
 
-  listContactsByUserId(): Promise<any> {
-    return new Promise((resolve, reject) => {
-      this.apiGateway
-        .get('contacts/list-contacts')
-        .subscribe((response: HttpResponse<any>) => {
-          resolve(response.body.apiResponse);
-        }, reject);
-    });
+
+  async listContactsByUserId(): Promise<ApiContactsResponse> {
+    let result = await this.apiClient.get('contacts/list-contacts');
+    return result
   }
 
-  saveNewContact(contact: any): Promise<any> {
-    return new Promise((resolve, reject) => {
-        this.apiGateway
-            .post('contacts/register', contact)
-            .subscribe((response: HttpResponse<any>) => {
-                resolve(response.body);
-            }, reject);
-    });
-}
+  async saveNewContact(contact: Contact): Promise<ApiContactsResponse> {
+    return await this.apiClient.post('contacts/register', contact);
+  }
 
-  updateContact(contact: any): Promise<any> {
-    return new Promise((resolve, reject) => {
-      this.apiGateway
-        .put('contacts/update', contact)
-        .subscribe((response: HttpResponse<any>) => {
-          resolve(response.body);
-        }, reject);
-    });
+  async updateContact(contact: Contact): Promise<ApiContactResponse> {
+    return await this.apiClient.put('contacts/update', contact);
   }
 
-  deleteContact(id: string | null): Promise<any> {
-    return new Promise((resolve, reject) => {
-        this.apiGateway
-            .delete('contacts/:id', { id: id })
-            .subscribe((response: HttpResponse<any>) => {
-                resolve(response.body);
-            }, reject);
-    });
-}
+  async deleteContact(id: string): Promise<ApiDeletedContact> {
+    const response = await this.apiClient.delete(`contacts`, { id: id });
+    return response.data;
```

Comment:
o apiClient já retorna o objeto data

```
async delete(url: string, params?: any) {
    let { data } = await axios.delete(`${environment.baseUrl}${url}/${params.id}`, this.getConfig());
    return data;
  }
```

este método deveria retornar apenas a resposta da chamada do apiClient, alterando para 

`return await this.apiClient.delete(`contacts`, { id: id });`

deveria funcionar normalmente, favor alterar e testar